app/src/main/java/com/github/gotify/sharing/ShareActivity.java

```diff
@@ -0,0 +1,96 @@
+package com.github.gotify.sharing;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import com.github.gotify.Settings;
+import com.github.gotify.Utils;
+import com.github.gotify.api.Api;
+import com.github.gotify.api.ApiException;
+import com.github.gotify.api.ClientFactory;
+import com.github.gotify.client.ApiClient;
+import com.github.gotify.client.api.ApplicationApi;
+import com.github.gotify.client.api.MessageApi;
+import com.github.gotify.client.model.Application;
+import com.github.gotify.client.model.Message;
+import com.github.gotify.log.Log;
+import java.util.List;
+
+import static com.github.gotify.Utils.first;
+
+public class ShareActivity extends Activity {
+    private ApiClient client;
+    private Settings settings;
+    private ShareActivity shareActivity;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        shareActivity = this;
+        settings = new Settings(this);
+        handleShareIntent();
+    }
+
+    private void handleShareIntent() {
+        Intent intent = getIntent();
+        String action = intent.getAction();
+        String type = intent.getType();
+
+        if (Intent.ACTION_SEND.equals(action) && type != null) {
+            if ("text/plain".equals(type)) {
+                String sharedText = intent.getStringExtra(Intent.EXTRA_TEXT);
+                if (sharedText != null) {
+                    Message message = new Message();
+                    message.setMessage(sharedText);
+                    message.setTitle("Shared content");
```

Comment:
Title and priority should be configurable. Currently gotify/android doesn't have a settings page, so a new page should be added.